js/ui/file_manager/ui.file_manager.item_list.js

```diff
@@ -1,13 +1,45 @@
 import { extend } from "../../core/utils/extend";
+import dblclickEvent from "../../events/double_click";
+import eventUtils from "../../events/utils";
+import eventsEngine from "../../events/core/events_engine";
 
+import ContextMenu from "../context_menu/ui.context_menu";
 import Widget from "../widget/ui.widget";
 
 const FILE_MANAGER_FILES_VIEW_CLASS = "dx-filemanager-files-view";
+const FILE_MANAGER_ITEM_LIST_ITEM_OPEN_EVENT_NAMESPACE = "dxFileManager_open";
 
 class FileManagerItemListBase extends Widget {
 
     _initMarkup() {
         this.$element().addClass(FILE_MANAGER_FILES_VIEW_CLASS);
+
+        const dblClickEventName = eventUtils.addNamespace(dblclickEvent.name, FILE_MANAGER_ITEM_LIST_ITEM_OPEN_EVENT_NAMESPACE);
+        eventsEngine.on(this.$element(), dblClickEventName, this._getItemSelector(), this._onItemDblClick.bind(this));
+
+        super._initMarkup();
+    }
+
+    _displayContextMenu(element, offsetX, offsetY) {
+        this._contextMenu.option({
+            "target": element,
+            "position": {
+                of: element,
+                offset: offsetX + " " + offsetY
+            }
+        });
+        this._contextMenu.show();
+    }
+
+    _ensureContextMenu() {
+        if(this._contextMenu) {
+            return;
+        }
+
+        this._contextMenu = this._createComponent("<div>", ContextMenu, {
```

Comment:
It would better to render widget at the element attached to DOM

Reply:
Should I use the `this._createComponent($("<div>", ...)` construction?

Reply:
You can make it like below

```js
const $menu = $("<div>").appendTo(this.$element());
this._contextMenu = this._createComponent($menu, ContextMenu, {
```

Reply:
Changed to this version